platform/binance/transaction.go

```diff
@@ -1,190 +1,167 @@
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 	"github.com/trustwallet/blockatlas/pkg/logger"
 	"github.com/trustwallet/blockatlas/pkg/numbers"
+	"strconv"
 	"strings"
 	"sync"
-
-	"github.com/trustwallet/blockatlas/coin"
 )
 
 func (p *Platform) GetTxsByAddress(address string) (blockatlas.TxPage, error) {
-	// Endpoint supports queries without token query parameter
-	return p.GetTokenTxsByAddress(address, p.Coin().Symbol)
+	return p.GetTokenTxsByAddress(address, "")
 }
 
-func (p *Platform) GetTokenTxsByAddress(address string, token string) (blockatlas.TxPage, error) {
-	srcTxs, err := p.client.GetTxsOfAddress(address, token)
+func (p *Platform) GetTokenTxsByAddress(address, token string) (blockatlas.TxPage, error) {
+	txPage, err := p.explorerClient.GetTxsOfAddress(address, token)
 	if err != nil {
 		return nil, err
 	}
-	txs, err := p.getTxChildChan(srcTxs.Txs)
+	txs, err := p.getTxChildChan(txPage.Txs)
 	if err != nil {
 		return nil, err
 	}
-	return NormalizeTxs(txs, token, address), nil
-}
 
-// getTxChildChan get all child assets from a tx
-func (p *Platform) getTxChildChan(srcTxs []Tx) ([]Tx, error) {
-	txs := make([]Tx, 0)
-	var wg sync.WaitGroup
-	out := make(chan Tx, len(srcTxs))
-	for _, srcTx := range srcTxs {
-		if srcTx.HasChildren != 1 {
-			// Return the same transaction if doesn't have a child
-			txs = append(txs, srcTx)
+	var normTxs []blockatlas.Tx
+	for _, srcTx := range txs {
+		tx, ok := NormalizeTx(srcTx, address, token)
+		if !ok {
 			continue
 		}
-		wg.Add(1)
-		go func(srcTx Tx, out chan Tx, wg *sync.WaitGroup) {
-			defer wg.Done()
-			tx, err := p.client.GetTx(srcTx.Hash)
-			if err != nil {
-				// Return the same transaction if an error occurs
-				out <- srcTx
-				logger.Error("GetTransactionsByBlockChan", err, logger.Params{"hash": srcTx.Hash})
-				return
-			}
-			out <- tx
-		}(srcTx, out, &wg)
+		normTxs = append(normTxs, tx...)
 	}
-	wg.Wait()
-	close(out)
-	for r := range out {
-		txs = append(txs, r)
-	}
-	return txs, nil
+
+	return normTxs, nil
 }
 
-func normalizeTransfer(tx blockatlas.Tx, srcTx Tx, token, address string) (blockatlas.TxPage, bool) {
-	// Verify if the tx has more them one asset
-	if srcTx.HasChildren == 1 {
-		txs := make(blockatlas.TxPage, 0)
-		// Parse all assets as a transaction
-		for _, subTx := range srcTx.SubTxsDto.SubTxDtoList.getTxs() {
-			// If this is not called from a block observer_test, only get the user txs/assets
-			if !subTx.containAddress(address) {
-				continue
-			}
-			// Recursive call to normalize the tx
-			newTxs, ok := normalizeTransfer(tx, subTx, token, address)
-			if !ok {
-				continue
+func (p *Platform) getTxChildChan(srcTxs []DexTx) ([]DexTx, error) {
+	var (
+		wg      sync.WaitGroup
+		hashChan = make(chan TxHashRPC)
+	)
+
+	for i, srcT := range srcTxs {
+		if srcT.HasChildren == 1 {
+			wg.Add(1)
+			go func(srcTx DexTx, out chan TxHashRPC, wg *sync.WaitGroup) {
+				defer wg.Done()
+				defer close(out)
```

Comment:
If I'm not mistaken, panic should happen here, because you close the channel in one procedure, but in other programs you continue to write to it.
Solution - close the channel after `wg.Wait()`
It is also worth testing this func (with `-race` flag too)